src/schema/v2/homeView/sections/index.ts

```diff
@@ -43,6 +43,7 @@ export type HomeViewSection = {
   }
   ownerType?: OwnerType
   requiresAuthentication: boolean
+  shouldBeDisplayed?: (context: ResolverContext) => boolean
```

Comment:
Should we anticipate this function, if defined, to sometimes involve some async logic? Maybe we can handle that conditionally if the return value is a `Promise` and allow teams to decide between sync or async conditions?

```suggestion
  shouldBeDisplayed?: (context: ResolverContext) => boolean | Promise<boolean>
```

Reply:
Ah, good idea 👍🏽 (though I'll prob merge this and update after KS in the interest of time)

Reply:
Actually I took a quick stab, but adding async behavior needed some more careful consideration when computing the list of sections from the default zone. Let's def come back to this.